crates/vm2/src/instruction_handlers/far_call.rs

```diff
@@ -184,40 +181,29 @@ pub(crate) fn get_far_call_calldata<T, W>(
 
     match FatPointerSource::from_abi(raw_source) {
         FatPointerSource::ForwardFatPointer => {
-            if !is_pointer || pointer.offset > pointer.length || already_failed {
+            if !is_pointer || pointer.offset > pointer.length {
                 return None;
             }
 
             pointer.narrow();
         }
         FatPointerSource::MakeNewPointer(target) => {
-            if let Some(bound) = pointer.start.checked_add(pointer.length) {
-                if is_pointer || pointer.offset != 0 || already_failed {
-                    return None;
-                }
-                match target {
-                    FatPointerTarget::ToHeap => {
-                        grow_heap::<_, _, Heap>(&mut vm.state, bound).ok()?;
-                        pointer.memory_page = vm.state.current_frame.heap;
-                    }
-                    FatPointerTarget::ToAuxHeap => {
-                        grow_heap::<_, _, AuxHeap>(&mut vm.state, bound).ok()?;
-                        pointer.memory_page = vm.state.current_frame.aux_heap;
-                    }
+            if is_pointer || pointer.offset != 0 {
+                return None;
+            }
+            let bound = pointer
+                .start
+                .checked_add(pointer.length)
+                .unwrap_or(u32::MAX);
```

Comment:
Am I missing something, or would `saturating_add` work here as well? If it would, it looks more idiomatic.

Reply:
True, though no longer relevant as in the final version I had to differentiate between overflow happening or not.